aspnet/fundamentals/middleware.rst

```diff
@@ -1,8 +1,176 @@
-.. include:: /../common/stub-topic.txt
+Middleware
+==========
+By `Steve Smith`_
 
-|stub-icon| Middleware
-======================
+:doc:`owin`, the Open Web Interface for .NET, describes small application components that can be incorporated into an HTTP request pipeline as middleware. ASP.NET 5 has integrated support for middleware, which are wired up in an application's ``Configure`` method during :doc:`startup`.
 
-.. include:: /../common/stub-notice.txt
+In this article:
+	- `What is middleware`_
+	- `Creating a middleware pipeline with IApplicationBuilder`_
+	- `Built-in middleware`_
+	- `Writing middleware`_
 
-.. _issue: https://github.com/aspnet/Docs/issues/109
+`Download sample from GitHub <https://github.com/aspnet/docs/aspnet/fundamentals/middleware/_static/sample>`_. 
+
+What is middleware
+------------------
+
+The `OWIN specification <http://owin.org/spec/spec/owin-1.0.0.html#Definition>`_ defines middleware as:
```

Comment:
The middleware model in ASP.NET 5 really is completely independent of OWIN. We support OWIN through an adapter layer. The core middleware model in ASP.NET 5 isn't based on OWIN at all.